### ISSUE TITLE
[BUG] Customization of ConfigMap for Control Plane Metrics Not Reflected in Azure Managed Prometheus

### ISSUE THREAD

**Issue by aritraghosh** [Azure/AKS#4689]
**Describe the bug**
Any change in the configmap for editing metrics for control plane components in Azure Managed Prometheus doesn't take effect. This was due to a recent roll out a new version of the managed Prometheus data collector in CCP (AKS controlplane) as an update for collecting metrics. Unfortunately, we’ve identified a bug in this version that prevents the collector from honoring/accepting changes made by customers in their ConfigMap. As a result, you may currently only see the default minimal metrics for your control plane targets. This impacts only control plane monitoring.

The configmap changes will not take effect whether or not they were applied recently or in the past 

At this time, there is no workaround for this issue. Our team is committed to resolving this as quickly as possible and will update the issue with details

**To Reproduce**
1. [Enable Control Plane metrics in Managed Prometheus](https://learn.microsoft.com/en-us/azure/aks/control-plane-metrics-monitor#enable-control-plane-metrics-on-your-aks-cluster)
2.  Add/delete metrics through c[ustomization](https://learn.microsoft.com/en-us/azure/aks/control-plane-metrics-monitor#customize-control-plane-metrics) in the metric
3. The changes don't reflect in the metrics that are sent to the Azure monitor workspace 

**Expected behavior**
The metrics changes made in the [configmap](https://github.com/Azure/prometheus-collector/blob/89e865a73601c0798410016e9beb323f1ecba335/otelcollector/configmaps/ama-metrics-settings-configmap.yaml) should have been emitted to the Azure monitor workspace

**Comment by grzesuav** [comment]
@aritraghosh how I can check if I am affected? I have ongoing ticket with Azure support which takes now like three weeks where I am complain of missing controlplane metrics in one cluster (present in other clusters) but returning some data in few points of time (see attached image) - all I got it "everything seems ok on AKS and Managed Prometheus side" but if Managed Prometheus has some issues with applying user configuration it could explain that - however my config was not changed of many months

![Image](https://github.com/user-attachments/assets/075764ad-1f25-4b44-bf05-124023859038)

**Comment by aritraghosh** [comment]
@grzesuav Please share the support case for further investigation

**Comment by grzesuav** [comment]
@aritraghosh `TrackingID#2411050050004147`

**Comment by vishiy** [comment]
@grzesuav - I just checked and i see your azure monitor workspace/AMW (monitor-workspace) is above quota since 11/11. You can look at current usage under "metrics" menu in AMW UX in the azure portal. I see at times you go to 2.8 million samples/min and currently quota for your AMW is default/1 million. You can request for quota increase thru "help-->support + troubleshooting" menu in in AMW UX in the azure portal. Please request increase for both samples/min & time-series for your AMW.  This explains why data is spotty from time to time, due to throttling.

**Comment by grzesuav** [comment]
@vishiy I was looking at that but in portal it does not concerning, however I had some questions about reliability and visibility of the data, probably will open another issue for that. To exclude this as a reason I requested an increase of quota.

**Comment by grzesuav** [comment]
Hi, after all it seems to me I am affected by this and it is causing overcharging us by storage space used for metrics - https://github.com/Azure/AKS/issues/4696#issuecomment-2565530725 as explained in other post, it seems mine `minimalIngestionProfile` setting is not respected, which is causing a big overhead in ingested samples and quota issues.

@aritraghosh how the roadmap to fix this looks like ?

**Comment by ilaylow** [comment]
@aritraghosh Does this bug relate to why I'm not seeing all metrics part of the minimal ingestion profile for the `controlplane-kube-scheduler` exporter? 

E.g. I'm not observing `scheduler_pod_scheduling_duration_seconds` even though that's stated to be part of the minimal ingestion profile [here](https://learn.microsoft.com/en-us/azure/azure-monitor/containers/prometheus-metrics-scrape-configuration-minimal).

**Comment by aritraghosh** [comment]
@ilaylow  Have you set the target true in the configmap [here](https://github.com/Azure/prometheus-collector/blob/main/otelcollector/configmaps/ama-metrics-settings-configmap.yaml#L29). kube-scheduler is not enabled by default.

**Comment by ilaylow** [comment]
@aritraghosh Yes I have, but the issue is I'm only seeing a subset of the scheduler metrics. Could it have something to do with the AKS version we're running?

![Image](https://github.com/user-attachments/assets/8a2a7b35-c964-4d4e-a6f8-066f4fd91158)

**Comment by aritraghosh** [comment]
The fix for the original issue has started rolling out. It will complete rollout in ~2 weeks. I will post an update here once its rolled out completely

**Comment by aritraghosh** [comment]
@ilaylow  It is possible those metrics from scheduler are not generated from the kube-scheduler for your configuration.

**Comment by ilaylow** [comment]
Thanks for the update @aritraghosh.
